src/backend/commands/trigger.c

```diff
@@ -2571,11 +2572,13 @@ ExecCallTriggerFunc(TriggerData *trigdata,
 	MyTriggerDepth++;
 	PG_TRY();
 	{
+		triggerOids = lappend_oid(triggerOids, trigdata->tg_trigger->tgoid);
 		result = FunctionCallInvoke(fcinfo);
 	}
 	PG_FINALLY();
 	{
 		MyTriggerDepth--;
+		triggerOids = list_delete_oid(triggerOids, trigdata->tg_trigger->tgoid);
```

Comment:
what if the tgoid duplicate in the list ?

Reply:
I can do one of two things
1. Delete last element from this list ( Nee to see if it is always trigdata->tg_trigger->tgoid)
2. Search last occurance of trigdata->tg_trigger->tgoid in this list and delete that